python/shotgun_model/shotgun_model.py

```diff
@@ -1397,6 +1483,24 @@ def _generate_display_name(self, field, sg_data):
             # everything else just cast to string
             return str(value)
 
+    def __get_columns(self, item, is_leaf):
+        """
+        Returns a row (list of QStandardItems) given an initial QStandardItem.  The item itself
+        is always the first item in the row, but additional columns may be appended.
+
+        :param item: A :class:`~PySide.QtGui.QStandardItem` that is associated with this model.
+        :param is_leaf: A boolean indicating if the item is a leaf item or not
+
+        :returns: A list of :class:`~PySide.QtGui.QStandardItem`s
+        """
+        # the first item in the row is always the standard shotgun model item,
+        # but subclasses may provide additional columns to be appended.
+        row = [item]
+        additional_columns = self._get_additional_columns(item, is_leaf, self.__column_fields)
+        if additional_columns:
+            row += additional_columns
```

Comment:
Or even the whole thing with:

`return [item].extend(self._get_additional_columns(item, ...))`